bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GenericItem.java

```diff
@@ -408,6 +408,11 @@ public StateDescription getStateDescription(Locale locale) {
                     result = stateDescription;
                 }
 
+                if (result != null && stateDescription != null) {
```

Comment:
You are here in any case overwriting a previous readOnly information, I don't think this is a valid approach.